src/histolab/mixins.py

```diff
@@ -0,0 +1,44 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------
+# Copyright 2021 All Histolab Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import numpy as np
+
+
+class LinalgMixin:
+    @staticmethod
+    def normalize_columns(arr: np.ndarray) -> np.ndarray:
+        """Normalize each column vector in an array"""
+        return arr / np.linalg.norm(arr, axis=0)
+
+    @staticmethod
+    def two_principal_components(arr: np.ndarray) -> np.ndarray:
```

Comment:
it would make sense, but I cannot seem to make it work with the "canonical way of doing PCA with eigenvectors decomposition", something like this

```python
eigen_values, eigen_vectors = np.linalg.eigh(np.cov(arr, rowvar=False))
sorted_index = np.argsort(eigen_values)[::-1]
sorted_eigenvectors = eigen_vectors[:, sorted_index]

return sorted_eigenvectors[:, 0:2]
```

[staintools](https://github.com/Peter554/StainTools/blob/master/staintools/stain_extraction/macenko_stain_extractor.py#L33) and [torchstain](https://github.com/EIDOSlab/torchstain/blob/main/torchstain/normalizers/numpy_macenko_normalizer.py#L29) are just getting the two middle columns,
while [histomicstk](https://github.com/DigitalSlideArchive/HistomicsTK/blob/588ecc7b96a5fc2a41b88f80115c780ab26e4edc/histomicstk/preprocessing/color_deconvolution/_linalg.py#L9) is using SVD, and for that specific input they don't have to slice anything


so... honestly I just took what worked and in this situation I don't know how to generalize this

Reply:
yes we are slicing but not how one would expect 😅 donno

Reply:
in practice, to compute the PCs there are several methods, including using the covariance matrix or the singular value decomposition (svd):

- case 1 i.e. we use the covariance matrix: it turns out that ```np.linalg.eigh``` returns the eigenvalues  in order of increasing magnitude, thus the two principal eigenvectors are the last ones, meaning that either we sort the columns and take the first 2 or we take the last two (as done by staintools). 

- case 2  i.e. we use the svd approach: in this case, we should first transpose our arr and then apply the svd, this would give us directly the PC matrix, but we have to slide it anyway (as done by [histomicstk](https://github.com/DigitalSlideArchive/HistomicsTK/blob/7860903b44dd564190d2234e475b7516af136127/histomicstk/preprocessing/color_deconvolution/separate_stains_macenko_pca.py#L85) ) to take the first two.

I think the second approach is preferable, as we are working with 2-stain images we do not need more components, however we can parametrize this as suggested by @ernestoarbitrio